### PR TITLE
Upgrade dependencies - only lib

### PR DESCRIPTION
**Pull request by vttranlina**
The pr https://github.com/apache/james-project/pull/1647 has a lot of files changed. 
The ci can't green after several re-runs, So this pr is a split part of 1674 for easier

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -619,7 +619,7 @@
         <junit.platform.version>1.9.3</junit.platform.version>
         <junit.vintage.version>5.9.3</junit.vintage.version>
         <concurrent.version>1.3.4</concurrent.version>
-        <xbean-spring.version>4.23</xbean-spring.version>
+        <xbean-spring.version>4.21</xbean-spring.version>
```
Comment (Arsnael) [inline pom.xml:622]: Anything wrong with that?

Comment (vttranlina) [reply pom.xml:622]: CI fail at Spring module
https://ci-builds.apache.org/blue/organizations/jenkins/james%2FApacheJames/detail/PR-1655/2/tests/

```
Error creating bean with name 'serverConnector' defined in class path resource [META-INF/org/apache/james/spring-server.xml]: Invocation of init method failed; nested exception is java.io.IOException: Cannot bind to URL [rmi://127.0.0.1:9999/jmxrmi]: javax.naming.CommunicationException [Root exception is java.rmi.ServerException: RemoteException occurred in server thread; nested exception is: 
 java.rmi.UnmarshalException: error unmarshalling arguments; nested exception is: 
 java.io.InvalidClassException: filter status: REJECTED]
Stacktrace
org.springframework.beans.factory.BeanCreationException: 
Error creating bean with name 'serverConnector' defined in class path resource [META-INF/org/apache/james/spring-server.xml]: Invocation of init method failed; nested exception is java.io.IOException: Cannot bind to URL [rmi://127.0.0.1:9999/jmxrmi]: javax.naming.CommunicationException [Root exception is java.rmi.ServerException: RemoteException occurred in server thread; nested exception is: 
	java.rmi.UnmarshalException: error unmarshalling arguments; nested exception is: 
	java.io.InvalidClassException: filter status: REJECTED]
	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:26)
Caused by: java.io.IOException: 
Cannot bind to URL [rmi://127.0.0.1:9999/jmxrmi]: javax.naming.CommunicationException [Root exception is java.rmi.ServerException: RemoteException occurred in server thread; nested exception is: 
	java.rmi.UnmarshalException: error unmarshalling arguments; nested exception is: 
	java.io.InvalidClassException: filter status: REJECTED]
	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:26)
Caused by: javax.naming.CommunicationException
	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:26)
Caused by: java.rmi.ServerException: 
RemoteException occurred in server thread; nested exception is: 
	java.rmi.UnmarshalException: error unmarshalling arguments; nested exception is: 
	java.io.InvalidClassException: filter status: REJECTED
	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:26)
Caused by: java.rmi.UnmarshalException: 
error unmarshalling arguments; nested exception is: 
	java.io.InvalidClassException: filter status: REJECTED
Caused by: java.io.InvalidClassException: filter status: REJECTED
```

**Comment by vttranlina** [comment]
CI error
```
setFlagsShouldWorkWithConcurrencyWithRemove – org.apache.james.mailbox.cassandra.mail.CassandraMessageMapperTest

Error
Expecting empty but was: ["custom-3-9"]
Stacktrace
java.lang.AssertionError: 
Expecting empty but was: ["custom-3-9"]
Standard Output
08:27:05.879 [ERROR] o.a.j.m.c.m.CassandraMessageMapper - Can not update following UIDs [ComposedMessageId{mailboxId=CassandraId{id=33f35700-2b8e-11ee-8d16-b744774e0118}, messageId=CassandraMessageId{uuid=33f552d0-2b8e-11ee-8d16-b744774e0118}, uid=MessageUid{uid=1}}] for mailbox 33f35700-2b8e-11ee-8d16-b744774e0118
```

It is not related to other commits in this pr. 
test cases: `setFlagsShouldWorkWithConcurrencyWithRemove` and `userFlagsUpdateShouldWorkInConcurrentEnvironment` (concurrency test) is unstable (exactly is it not work, but the `threadCount` and `updateCount` is quite small, so it is not enough to easier make the test fail)
=> I marked `@Disabled` it

**Comment by chibenwa** [comment]
Tag it as unstable instead please

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by chibenwa** [comment]
Quock perf test maybe?

**Pull request merged**